EIPS/eip-770.md

```diff
@@ -0,0 +1,124 @@
+
+(Step 6.1 - Optional) Booker cancels the reservation by broadcasting a cancellation request. This triggers a submission to the RES smart contract that empties the registries for this resource.
+Depending on the conditions and cancellation date, the BTU escrowed into the RES smart contract are affected back to the user or to the provider
+
+(Step 6.2 - Optional) Provider cancels the reservation by broadcasting a cancellation request. This triggers a submission to the RES smart contract that empties the registries for this resource. RES smart contract is releasing the escrowed amount to booker.
+
+(Step 7.1 & 7.2) Booker is notifying that the resource has been paid (presumably at resource check-out). RES smart contract is releasing the escrowed BTU back to user in addition to a BTU agreed commision.
+
+## Implementation
+
+The implementation involves following functions :
+
+- publishAvailabilities (likely off-chain) : Publishes one or multiple available resources for reservations and for being “searchable”.
+- listAvailabilities (likely off-chain) : List all available resources matching a search criteria
+- requestReservation : Request for a reservation. The function can be used for initial reservation request OR updating
+- getReservationStatus : Read reservation status. Following values are possible : REQUESTED, REJECTED, CONFIRMED
+
+
+```
+contract ERC770 is ERC20 {
+    // Availability structure
+    enum BookingStatus { REQUESTED, REJECTED, CONFIRMED, CANCELLED  }
+    struct availability {
+	address               _contractAdress;
+	uint                     _resourceId ;   // resource id OR bundle id
+	uint                     _type;   // Type of Availability
+	uint                     _minDeposit ; // minimum BTU deposit for booking this resource
+	uint                     _commission ; // commission amount paid to booker in BTU
+	uint                     _freeCancelDateTs; // Limit date for a reservation cancellation
+	uint                     _statDateTs;   //availability start date timestamps
+	uint                     _endDateTs;   //availability end date timestamps
+	BookingStatus    _bookingStatus ; // reservation status
+	string                  _metaDataLink // Link to Meta Data of the bookable resource (desc, image links, etc…)
+   }
+    //Submit one or multiple availability - implementation will be off-chain
+    function publishAvailabilities (address _owner, availability[] _availability, bytes32 signatureProof )  constant returns (uint status);
+    //Query Availabilities - implementation will be off-chain
+    function ListAvailabilities(address _requester, string _criterias)  constant returns (availability[] a);
+    //Request reservation
+    function requestReservation(address _requester, availability _availability)  constant returns (uint status);
+    //Check booking status
+    function getReservationStatus(address _requester, availability _availability)  constant returns (BookingStatus status);
```

Comment:
Why is `status` sometimes `BookingStatus` and sometimes `uint`?

Reply:
Thanks for this relevent remark. The correct type is BookingStatus and this mistake came from a bad copy and paste (from a previous code base)